File: 백준/Silver/2164. 카드2/카드2.js
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(path).toString().trim();

const N = Number(input);

const queue = Array(N)
  .fill()
  .map((v, i) => i + 1);

let front = 0;
while (queue.length - front > 1) {
  front++;
  queue.push(queue[front]);
  front++;
}

console.log(queue[front]);

File: 백준/Silver/2164. 카드2/카드2.py
import sys
from collections import deque 

input = sys.stdin.readline
N = int(input())
queue = deque([i + 1 for i in range(N)])

while len(queue) > 1:
  queue.popleft()
  if queue:
    queue.append(queue.popleft())

print(queue[0])



